### PR TITLE
Remote config

### PR DESCRIPTION
**Pull request by rock3r**
This PR adds remote config facilities so that we can control remotely whether the proximity services are enabled.

### CODE REVIEW

File: app/src/main/java/net/squanchy/remoteconfig/RemoteConfig.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    public Single<Boolean> proximityServicesEnabled() {
+        return getBooleanConfigValue(() -> remoteConfig.getBoolean(KEY_PROXIMITY_ENABLED))
+                .subscribeOn(Schedulers.io());
+    }
+
+    private <T> Single<T> getBooleanConfigValue(Func0<T> action) {
+        return Single.create(emitter ->
+                remoteConfig.fetch(cacheExpiryInSeconds())
+                        .addOnCompleteListener(task -> {
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/remoteconfig/RemoteConfig.java:35]: I'd add an on error listener too

**Review by fourlastor — APPROVED** [review]
Seems fair enough apart from adding a listener, is there any way to make this synchronous though?

File: app/src/main/java/net/squanchy/remoteconfig/RemoteConfig.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/remoteconfig/RemoteConfig.java:35]: I'm following the example code and docs here, looks like it's the `task.isSuccessful()` that tells us about errors. Which, I don't really care about anyway, at least for now I'm happy with the defaults in that case.

Comment (fourlastor) [reply app/src/main/java/net/squanchy/remoteconfig/RemoteConfig.java:35]: 🚢 then

**Comment by rock3r** [comment]
@fourlastor no way to have a synchronous call unfortunately, it's a weird and convoluted API. Just checked with Luis V. and it's what they do in their app too :P

**Pull request merged**